### ISSUE TITLE
Button collections do not toggle/untoggle when currentIndex changes.

### ISSUE THREAD

**Issue by FlyingFlambe** [microsoft/MixedRealityToolkit-Unity#6274]
## Describe the bug

TL;DR - Pressing a button in a collection will update the currentIndex, but changing the currentIndex does not change the toggle status of buttons in the collection.

A button can be toggled on by selecting a button from the index of buttons, but a different button in that collection cannot be toggled on if the currentIndex value is manually changed in the script or the inspector.

## To reproduce

1. Create a button collection with as many buttons as you'd like (maybe 5).
2. Make sure all buttons are toggle-able so a button stays pressed after being clicked on.
3. In Play Mode, click on the various buttons while observing the Inspector. The currentIndex should change appropriately.
4. BUG: While still in Play Mode, use the Inspector and change the currentIndex value. The corresponding button should be toggled on, but it does not. (Play Mode can be stopped after observing this.)
5. BUG: If a script is written so as to change the currentIndex value, the corresponding button won't be toggled.

## Expected behavior

If the currentIndex value is changed, the button collection should toggle on/off the appropriate buttons that correspond to that index value.

## Your Setup (please complete the following information)

- Unity 2019.1.14f1
- MRTK Version RC2.1

## Target Platform (please complete the following information)

- HoloLens

## Additional context

Also noticed that with any MRTK button, it needs to be explicitly pressed by the user for the button press to work. There's no function to simulate a button press within a script. As a UX Designer & Dev, both the button collections fix and being able to simulate a button press would help a ton.

**Comment by cre8ivepark** [comment]
@killerantz Could you please share information on this?

**Comment by davidkline-ms** [comment]
@CDiaz-MS, can you confirm whether or not this repros on 2.1.0?

**Comment by FlyingFlambe** [comment]
@davidkline-ms @CDiaz-MS It reproduced on 2.1; can be tested by using the Radial Set prefab.

**Comment by Troy-Ferrell** [comment]
I can repo as well. InteractableToggleCollection does not have any get/set methods on the CurrentIndex property

**Comment by CDiaz-MS** [comment]
I can repro as well.  Working on a fix for this.
